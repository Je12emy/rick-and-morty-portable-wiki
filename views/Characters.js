import React from 'react';
import {SafeAreaView, StyleSheet, Text, FlatList, View} from 'react-native';
import {useQuery, useInfiniteQuery} from 'react-query';
import {ItemCard} from '../components/ItemCard';
import {useNavigation} from '@react-navigation/native';
import {normalizeData} from '../utils/normalizeData';
import {Searchbar} from 'react-native-paper';

export const Characters = () => {
  const navigation = useNavigation();
  // Search bar state
  const [searchQuery, setSearchQuery] = React.useState('');
  const [isEmpty, setIsEmpty] = React.useState(false);

  // When the search box is empty and the empty flag is true
  React.useEffect(() => {
    if (isEmpty) {
      if (searchQuery === '') {
        handleSearch();
        setIsEmpty(false);
      }
    }
  }, [searchQuery]);

  const onChangeSearch = (query) => {
    setSearchQuery(query);
    if (searchQuery === '') {
      setIsEmpty(true);
    }
  };

  const handleSearch = () => {
    // Remove all stored data
    data.pages.pop();
    fetchNextPage({pageParam: 1});
  };

  const {isLoading, error, data, fetchNextPage, hasNextPage} = useInfiniteQuery(
    'characters',
    async ({pageParam = 1}) => {
      console.info(
        'Query: ',
        'https://rickandmortyapi.com/api/character/?page=' +
          pageParam +
          '&name=' +
          searchQuery,
      );
      const request = await fetch(
        'https://rickandmortyapi.com/api/character/?page=' +
          pageParam +
          '&name=' +
          searchQuery,
      );
      if (!request.ok) {
        throw new Error('Network response was not ok');
      }
      const response = await request.json();
      return response;
    },
    {
      getNextPageParam: (lastPage, pages) => {
        if (lastPage !== undefined) {
          if (lastPage.info.next !== null) {
            return lastPage.info.next.slice(48);
          } else {
            return undefined;
          }
        } else {
          return undefined;
        }
      },
    },
  );

  const getMore = () => {
    fetchNextPage();
  };

  if (isLoading) return <Text>'Loading....'</Text>;

  if (error) return <Text>An error has occurred: {error.message}</Text>;

  const handleViewCharacter = (id) => {
    navigation.navigate('Character', {
      characterId: id,
    });
  };

  return (
    <SafeAreaView>
      <Searchbar
        style={{marginBottom: 1}}
        placeholder="Search"
        onChangeText={onChangeSearch}
        value={searchQuery}
        onIconPress={handleSearch}
      />
      <FlatList
        data={normalizeData(data)}
        renderItem={({item}) => (
          <ItemCard
            title={item.name}
            subtitle={item.species}
            image={item.image}
            id={item.id}
            onPress={handleViewCharacter}
          />
        )}
        keyExtractor={(character) => character.id}
        onEndReached={() => {
          if (hasNextPage) {
            getMore();
          }
        }}
        contentContainerStyle={{paddingBottom: 75}}
      />
    </SafeAreaView>
  );
};
