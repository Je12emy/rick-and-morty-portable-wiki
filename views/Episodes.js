import React from 'react';
import {SafeAreaView, StyleSheet, Text, FlatList} from 'react-native';
import {useQuery, useInfiniteQuery} from 'react-query';
import {ItemCard} from '../components/ItemCard';
import {useNavigation} from '@react-navigation/native';
import {normalizeData} from '../utils/normalizeData';
import {Searchbar} from 'react-native-paper';

export const Episodes = () => {
  const navigation = useNavigation();

  // Search bar state
  const [searchQuery, setSearchQuery] = React.useState('');
  const [isEmpty, setIsEmpty] = React.useState(false);

  // When the search box is empty and the empty flag is true
  React.useEffect(() => {
    if (isEmpty) {
      if (searchQuery === '') {
        handleSearch();
        setIsEmpty(false);
      }
    }
  }, [searchQuery]);

  const onChangeSearch = (query) => {
    setSearchQuery(query);
    if (searchQuery === '') {
      setIsEmpty(true);
    }
  };

  const handleSearch = () => {
    data.pages.pop();
    fetchNextPage({pageParam: 1});
  };

  const {isLoading, error, data, fetchNextPage, hasNextPage} = useInfiniteQuery(
    'episodes',
    async ({pageParam = 1}) => {
      console.log('Fetch page: ', pageParam);
      const request = await fetch(
        'https://rickandmortyapi.com/api/episode/?page=' +
          pageParam +
          '&name=' +
          searchQuery,
      );
      if (!request.ok) {
        throw new Error('Network response was not ok');
      }
      // JSON era una promesa
      const response = await request.json();
      return response;
    },
    {
      getNextPageParam: (lastPage, pages) => {
        if (lastPage !== undefined) {
          if (lastPage.info.next !== null) {
            return lastPage.info.next.slice(46);
          } else {
            return undefined;
          }
        } else {
          return undefined;
        }
      },
    },
  );
  // Mover a utils
  const getMore = () => {
    // fetchNextPage({pageParam: nextPage});
    const fetchResult = fetchNextPage();
    console.log(fetchResult);
  };

  if (isLoading) return <Text>'Loading....'</Text>;

  if (error) return <Text>An error has occurred: {error.message}</Text>;

  const handleViewEpisode = (id) => {
    navigation.navigate('Episode', {
      episodeId: id,
    });
  };

  // console.log('Next Page?', hasNextPage);

  return (
    <SafeAreaView>
      <Searchbar
        style={{marginBottom: 1}}
        placeholder="Search"
        onChangeText={onChangeSearch}
        value={searchQuery}
        onIconPress={handleSearch}
      />
      <FlatList
        data={normalizeData(data)}
        renderItem={({item}) => (
          <ItemCard
            title={item.name}
            subtitle={item.air_date}
            onPress={handleViewEpisode}
            id={item.id}
            icon="television"
          />
        )}
        keyExtractor={(item) => item.id.toString()}
        onEndReached={() => {
          if (hasNextPage) {
            getMore();
          }
        }}
        contentContainerStyle={{paddingBottom: 75}}
      />
    </SafeAreaView>
  );
};
