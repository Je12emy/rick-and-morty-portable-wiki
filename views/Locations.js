import React from 'react';
import {SafeAreaView, StyleSheet, Text, FlatList} from 'react-native';
import {useQuery, useInfiniteQuery} from 'react-query';
import {ItemCard} from '../components/ItemCard';
import {useNavigation} from '@react-navigation/native';
import {normalizeData} from '../utils/normalizeData';
import {Searchbar} from 'react-native-paper';

export const Locations = () => {
  const navigation = useNavigation();

  // Search bar state
  const [searchQuery, setSearchQuery] = React.useState('');
  const [isEmpty, setIsEmpty] = React.useState(false);

  // When the search box is empty and the empty flag is true
  React.useEffect(() => {
    if (isEmpty) {
      if (searchQuery === '') {
        handleSearch();
        setIsEmpty(false);
      }
    }
  }, [searchQuery]);

  const onChangeSearch = (query) => {
    setSearchQuery(query);
    if (searchQuery === '') {
      setIsEmpty(true);
    }
  };

  const handleSearch = () => {
    data.pages.pop();
    fetchNextPage();
  };

  const {isLoading, error, data, fetchNextPage, hasNextPage} = useInfiniteQuery(
    'locations',
    async ({pageParam = 1}) => {
      const request = await fetch(
        'https://rickandmortyapi.com/api/location/?page=' +
          pageParam +
          '&name=' +
          searchQuery,
      );
      if (!request.ok) {
        throw new Error('Network response was not ok');
      }
      // JSON era una promesa
      const response = await request.json();
      return response;
    },
    {
      getNextPageParam: (lastPage, pages) => {
        if (lastPage !== undefined) {
          if (lastPage.info.next !== null) {
            return lastPage.info.next.slice(47);
          } else {
            return undefined;
          }
        } else {
          return undefined;
        }
      },
    },
  );

  const getMore = () => {
    fetchNextPage();
  };

  if (isLoading) return <Text>'Loading....'</Text>;

  if (error) return <Text>An error has occurred: {error.message}</Text>;

  const handleViewCharacter = (id) => {
    navigation.navigate('Location', {
      locationId: id,
    });
  };
  return (
    <SafeAreaView>
      <Searchbar
        style={{marginBottom: 1}}
        placeholder="Search"
        onChangeText={onChangeSearch}
        value={searchQuery}
        onIconPress={handleSearch}
      />
      <FlatList
        data={normalizeData(data)}
        renderItem={({item}) => (
          <ItemCard
            title={item.name}
            subtitle={item.type}
            image={item.image}
            onPress={handleViewCharacter}
            id={item.id}
            icon="map-marker"
          />
        )}
        keyExtractor={(character) => character.id}
        onEndReached={() => {
          if (hasNextPage) {
            getMore();
          }
        }}
        contentContainerStyle={{paddingBottom: 75}}
      />
    </SafeAreaView>
  );
};
