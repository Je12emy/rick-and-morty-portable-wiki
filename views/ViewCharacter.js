import * as React from 'react';
import {View, Text, Image, ScrollView, SafeAreaView} from 'react-native';
import {Avatar, Card, List, Caption, Paragraph} from 'react-native-paper';
import {useQuery} from 'react-query';

export const ViewCharacter = ({route, navigation}) => {
  const {characterId} = route.params;
  const [expanded, setExpanded] = React.useState(false);

  // Episodes list
  const [episodes, setEpisodes] = React.useState([]);

  const handlePress = async () => {
    setExpanded(!expanded);
    // If episode list is empty
    if (episodes.length === 0) {
      await getEpisodeList();
    }
  };

  const handleEpisodeVisit = (id) => {
    navigation.navigate('Episode', {
      episodeId: id,
    });
  };

  const {isLoading, error, data} = useQuery('character', async () => {
    const request = await fetch(
      `https://rickandmortyapi.com/api/character/${characterId}`,
    );
    if (!request.ok) {
      throw new Error('Network response was not ok');
    }
    // JSON era una promesa
    const response = await request.json();
    return response;
  });

  const getGenderIcon = (gender) => {
    if (gender === 'Male') {
      return 'gender-male';
    }
    if (gender === 'Female') {
      return 'gender-female';
    } else {
      return 'gender-non-binary';
    }
  };

  const getStatusIcon = (status) => {
    if (status === 'Alive') {
      return 'heart-pulse';
    }
    if (status === 'Dead') {
      return 'skull-outline';
    } else {
      return 'help';
    }
  };

  // name, episodeId
  const getEpisodeList = async () => {
    // If current array is empty
    if (episodes.length === 0) {
      // episode list of id's
      const idList = [];
      const episodeData = [];

      data.episode.forEach((episode) => {
        idList.push(episode.slice(40));
      });

      const episodeList = await fetchEpisodeList(idList);
      if (idList.length > 1) {
        // Shape data
        episodeList.forEach((episode) => {
          episodeData.push({
            name: episode.name,
            id: episode.id,
            episode: episode.episode,
          });
        });
      } else {
        episodeData.push(episodeList);
      }
      setEpisodes(episodeData);
    }
  };

  const fetchEpisodeList = async (idList) => {
    // Fetch episode list
    const request = await fetch(
      `https://rickandmortyapi.com/api/episode/${idList}`,
    );
    if (!request.ok) {
      throw new Error('Network response was not ok');
    }
    // JSON era una promesa
    const response = await request.json();
    return response;
  };

  if (isLoading) return <Text>'Loading....'</Text>;

  if (error) return <Text>An error has occurred: {error.message}</Text>;
  return (
    <SafeAreaView>
      <ScrollView>
        <View style={{justifyContent: 'center', alignItems: 'center'}}>
          <Avatar.Image
            size={200}
            source={{
              uri: data.image,
            }}
            style={{marginTop: 8, marginBottom: 8}}
          />
        </View>
        <View>
          <Paragraph style={{marginBottom: 8, marginLeft: 16}}>
            Information
          </Paragraph>
          <Card elevation={1} style={{marginBottom: 1}}>
            <Card.Title
              title="Name"
              subtitle={data.name}
              left={(props) => <Avatar.Icon {...props} icon="form-textbox" />}
            />
          </Card>
          <Card elevation={1} style={{marginBottom: 1}}>
            <Card.Title
              title="Gender"
              subtitle={data.gender}
              left={(props) => (
                <Avatar.Icon {...props} icon={getGenderIcon(data.gender)} />
              )}
            />
          </Card>
          <Card elevation={1} style={{marginBottom: 1}}>
            <Card.Title
              title="Species"
              subtitle={data.species}
              left={(props) => <Avatar.Icon {...props} icon="human" />}
            />
          </Card>
          <Card elevation={1} style={{marginBottom: 1}}>
            <Card.Title
              title="Status"
              subtitle={data.status}
              left={(props) => (
                <Avatar.Icon {...props} icon={getStatusIcon(data.status)} />
              )}
            />
          </Card>
          <Card elevation={1} style={{marginBottom: 1}}>
            <Card.Title
              title="Origin"
              subtitle={data.origin.name}
              left={(props) => <Avatar.Icon {...props} icon="earth" />}
            />
          </Card>
          <Card elevation={1} style={{marginBottom: 1}}>
            <Card.Title
              title="Location"
              subtitle={data.location.name}
              left={(props) => <Avatar.Icon {...props} icon="map-marker" />}
            />
          </Card>
          {
            // apply conditional
            data.type !== '' ? (
              <Card elevation={1} style={{marginBottom: 1}}>
                <Card.Title
                  title="Type"
                  subtitle={data.type}
                  left={(props) => (
                    <Avatar.Icon {...props} icon="account-alert" />
                  )}
                />
              </Card>
            ) : null
          }

          <List.Section title="Featured in">
            <List.Accordion
              title="List of Episodes"
              left={(props) => <List.Icon {...props} icon="folder" />}
              expanded={expanded}
              onPress={handlePress}>
              {
                // map episodes
                episodes.map((episode, index) => (
                  <List.Item
                    id={episode.id}
                    title={episode.name}
                    description={episode.episode}
                    left={(props) => <List.Icon {...props} icon="television" />}
                    onPress={() => {
                      handleEpisodeVisit(episode.id);
                    }}
                    key={episode.id.toString()}
                  />
                ))
              }
            </List.Accordion>
          </List.Section>
        </View>
      </ScrollView>
    </SafeAreaView>
  );
};
