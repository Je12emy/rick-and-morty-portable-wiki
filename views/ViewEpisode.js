import * as React from 'react';
import {View, Text, Image, ScrollView, SafeAreaView} from 'react-native';
import {Card, List, Caption, Paragraph, Avatar} from 'react-native-paper';
import {useQuery} from 'react-query';
import {imageIcon} from '../components/ItemCard';

export const ViewEpisode = ({route, navigation}) => {
  const {episodeId} = route.params;
  const [expanded, setExpanded] = React.useState(false);

  // Character list
  const [characters, setCharacters] = React.useState([]);

  const handlePress = async () => {
    setExpanded(!expanded);
    if (characters.length === 0) {
      await getCharacterList();
    }
  };

  const handleCharacterVisit = (id) => {
    navigation.navigate('Character', {
      characterId: id,
    });
  };

  const {isLoading, error, data} = useQuery('location', async () => {
    const request = await fetch(
      `https://rickandmortyapi.com/api/episode/${episodeId}`,
    );
    if (!request.ok) {
      throw new Error('Network response was not ok');
    }
    // JSON era una promesa
    const response = await request.json();
    return response;
  });

  const getCharacterList = async () => {
    // If current array is empty
    if (characters.length === 0) {
      // episode list of id's
      const idList = [];
      const characterData = [];

      data.characters.forEach((episode) => {
        idList.push(episode.slice(42));
      });

      const characterList = await fetchCharacterList(idList);
      if (idList.length > 1) {
        // Shape data
        characterList.forEach((character) => {
          characterData.push({
            name: character.name,
            id: character.id,
            origin: character.origin.name,
            image: character.image,
          });
        });
      } else {
        characterData.push(characterList);
      }

      setCharacters(characterData);
    }
  };

  const fetchCharacterList = async (idList) => {
    // Fetch character list
    const request = await fetch(
      `https://rickandmortyapi.com/api/character/${idList}`,
    );
    if (!request.ok) {
      throw new Error('Network response was not ok');
    }
    // JSON era una promesa
    const response = await request.json();
    return response;
  };

  if (isLoading) return <Text>'Loading....'</Text>;

  if (error) return <Text>An error has occurred: {error.message}</Text>;
  return (
    <SafeAreaView>
      <ScrollView>
        <View style={{marginTop: 8}}>
          <Paragraph style={{marginBottom: 8, marginLeft: 16}}>
            Information
          </Paragraph>
          <Card elevation={1} style={{marginBottom: 1}}>
            <Card.Title
              title="Name"
              subtitle={data.name}
              left={(props) => <Avatar.Icon {...props} icon="form-textbox" />}
            />
          </Card>
          <Card elevation={1} style={{marginBottom: 1}}>
            <Card.Title
              title="Air Date"
              subtitle={data.air_date}
              left={(props) => <Avatar.Icon {...props} icon="calendar-range" />}
            />
          </Card>
          <Card elevation={1} style={{marginBottom: 1}}>
            <Card.Title
              title="Episode"
              subtitle={data.episode}
              left={(props) => (
                <Avatar.Icon {...props} icon="format-list-numbered" />
              )}
            />
          </Card>
          <List.Section title="Featured Characters">
            <List.Accordion
              title="List of Characters"
              left={(props) => <List.Icon {...props} icon="folder" />}
              expanded={expanded}
              onPress={handlePress}>
              {characters.map((character) => (
                <List.Item
                  key={character.id}
                  title={character.name}
                  description={character.origin}
                  left={() => imageIcon(character)}
                  onPress={() => {
                    handleCharacterVisit(character.id);
                  }}
                />
              ))}
            </List.Accordion>
          </List.Section>
        </View>
      </ScrollView>
    </SafeAreaView>
  );
};
