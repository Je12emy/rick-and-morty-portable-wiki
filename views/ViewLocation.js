import * as React from 'react';
import {View, Text, Image, ScrollView, SafeAreaView} from 'react-native';
import {Avatar, Card, List, Caption, Paragraph} from 'react-native-paper';
import {useQuery} from 'react-query';
import {imageIcon} from '../components/ItemCard';

export const ViewLocation = ({route, navigation}) => {
  const {locationId} = route.params;
  const [expanded, setExpanded] = React.useState(true);
  // Character list
  const [characters, setCharacters] = React.useState([]);

  const handlePress = async () => {
    await getResidentList();
    setExpanded(!expanded);
  };

  const handleCharacterVisit = (id) => {
    navigation.navigate('Character', {
      characterId: id,
    });
  };

  const {isLoading, error, data} = useQuery('location', async () => {
    const request = await fetch(
      `https://rickandmortyapi.com/api/location/${locationId}`,
    );
    if (!request.ok) {
      throw new Error('Network response was not ok');
    }
    // JSON era una promesa
    const response = await request.json();
    return response;
  });

  const getResidentList = async () => {
    // If current array is empty
    if (characters.length === 0) {
      // episode list of id's
      const idList = [];
      const characterData = [];
      console.log(idList);
      data.residents.forEach((episode) => {
        idList.push(episode.slice(42));
      });

      const characterList = await fetchResidentList(idList);
      if (idList.length > 1) {
        // Shape data
        characterList.forEach((character) => {
          characterData.push({
            name: character.name,
            id: character.id,
            origin: character.origin.name,
            image: character.image,
          });
        });
      } else {
        characterData.push({
          name: characterList.name,
          id: characterList.id,
          origin: characterList.origin.name,
          image: characterList.image,
        });
      }
      setCharacters(characterData);
    }
  };

  const fetchResidentList = async (idList) => {
    // Fetch character list
    const request = await fetch(
      `https://rickandmortyapi.com/api/character/${idList}`,
    );
    if (!request.ok) {
      throw new Error('Network response was not ok');
    }
    // JSON era una promesa
    const response = await request.json();
    return response;
  };

  if (isLoading) return <Text>'Loading....'</Text>;

  if (error) return <Text>An error has occurred: {error.message}</Text>;
  return (
    <SafeAreaView>
      <ScrollView>
        <View style={{marginTop: 8}}>
          <Paragraph style={{marginBottom: 8, marginLeft: 16}}>
            Information
          </Paragraph>
          <Card elevation={1} style={{marginBottom: 1}}>
            <Card.Title
              title="Name"
              subtitle={data.name}
              left={(props) => <Avatar.Icon {...props} icon="form-textbox" />}
            />
          </Card>
          <Card elevation={1} style={{marginBottom: 1}}>
            <Card.Title
              title="Type"
              subtitle={data.type}
              left={(props) => <Avatar.Icon {...props} icon="dna" />}
            />
          </Card>
          <Card elevation={1} style={{marginBottom: 1}}>
            <Card.Title
              title="Dimension"
              subtitle={data.dimension}
              left={(props) => <Avatar.Icon {...props} icon="cube-outline" />}
            />
          </Card>

          <List.Section title="Featured Residents">
            <List.Accordion
              title="List of Residents"
              left={(props) => <List.Icon {...props} icon="folder" />}
              expanded={expanded}
              onPress={handlePress}>
              {characters.map((character) => (
                <List.Item
                  key={character.id}
                  title={character.name}
                  description={character.origin}
                  left={() => imageIcon(character)}
                  onPress={() => {
                    handleCharacterVisit(character.id);
                  }}
                />
              ))}
            </List.Accordion>
          </List.Section>
        </View>
      </ScrollView>
    </SafeAreaView>
  );
};
