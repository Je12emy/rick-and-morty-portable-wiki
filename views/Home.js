import React from 'react';
import {StyleSheet, Dimensions, StatusBar, Text, View} from 'react-native';
import {TabView, SceneMap, TabBar} from 'react-native-tab-view';
const initialLayout = {width: Dimensions.get('window').width};

import {Characters} from './Characters';
import {Episodes} from './Episodes';
import {Locations} from './Locations';

export const Home = ({theme}) => {
  const RenderTabBar = (props) => (
    <View>
      <StatusBar backgroundColor="#005005"/>
      <TabBar
        {...props}
        indicatorStyle={{backgroundColor: theme.colors.accent}}
        style={{backgroundColor: theme.colors.primary}}
        renderLabel={({route, focused, color}) => (
          <Text
            style={{
              margin: 8,
              color: color,
            }}>
            {route.title}
          </Text>
        )}
      />
    </View>
  );

  // Set up tabs
  const [index, setIndex] = React.useState(0);
  const [routes] = React.useState([
    {key: 'first', title: 'Characters'},
    {key: 'second', title: 'Locations'},
    {key: 'third', title: 'Episodes'},
  ]);
  const renderScene = SceneMap({
    first: Characters,
    second: Locations,
    third: Episodes,
  });
  return (
    <TabView
      navigationState={{index, routes}}
      renderScene={renderScene}
      onIndexChange={setIndex}
      initialLayout={initialLayout}
      renderTabBar={(props) => <RenderTabBar {...props} />}
    />
  );
};
