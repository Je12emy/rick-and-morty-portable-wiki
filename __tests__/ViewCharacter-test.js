import React from 'react';
import renderer from 'react-test-renderer';
import {ViewCharacter} from '../views/ViewCharacter';
import {QueryClient, QueryClientProvider} from 'react-query';

// const wrapper = ({children}) => {
//   <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>;
// };



// test('renders view character correctly', () => {
//   const queryClient = new QueryClient();

//   const tree = renderer
//     .create(
//       <QueryClientProvider client={queryClient}>
//         <ViewCharacter
//           route={{
//             params: {
//               characterId: 1,
//             },
//           }}
//         />
//       </QueryClientProvider>,
//     )
//     .toJSON();
//   expect(tree).toMatchSnapshot();
// });
