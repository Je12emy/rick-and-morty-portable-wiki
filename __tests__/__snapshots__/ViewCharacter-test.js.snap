// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders view character correctly 1`] = `
<Text>
  'Loading....'
</Text>
`;

exports[`renders view character view 1`] = `
<Text>
  'Loading....'
</Text>
`;
